### ISSUE TITLE
I keep getting a webSocket error.

### ISSUE THREAD

**Issue by angelmartesSIGMA** [titaniumnetwork-dev/Ultraviolet#168]
Error: Hyper client: hyper_util::client::legacy::Error(Connect, WebSocketConnectFailed("websocket did not open")) (client error (Wisp WebSocket failed to connect: websocket did not open))

How do I fix it?

**Comment by angelmartesSIGMA** [comment]
![Screenshot 2024-11-26 12 02 41 PM](https://github.com/user-attachments/assets/9080f27a-acca-4109-85de-986216dc6306)

**Comment by NottPeak** [comment]
That actually looks like an issue with Shadow, assuming you aren't selfhosting the link something got messed up with the wisp server it's trying to use.

try going into settings > developer and switch the drop-down on the left to custom. Enter wss://shadow.letsendobesity.org/wisp/ (or put the dropdown back to default) and try again (may need to clear your cache)

If the issue persists, you can join our [discord](https://discord.gg/goshadow) and create a ticket for support

**Comment by angelmartesSIGMA** [comment]
> That actually looks like an issue with Shadow, assuming you aren't selfhosting the link something got messed up with the wisp server it's trying to use.

i dont even think I have a wisp server, I don't really know how to set up a wisp server and I thought it would set one up automatically

I am using a VPS run on Ubuntu 22.04, so it's not really self-hosting I think I dunno

**Comment by NottPeak** [comment]
> hosting shadow on your own server is considered self-hosting, and they are shipping a wisp server with it but it's not being used for some reason
> 
> I'm getting the same thing when I try to launch shadow on my device - this is an error on their side, so I think it's best if you open an issue on their [github repository](https://github.com/shadowdevlabs/shadowv3) or in their discord as Peak mentioned

I'll take a look at it tonight, our production environment runs the python implementation of wisp so we could have messed something up with the node server and not noticed it

**Comment by NottPeak** [comment]
> > That actually looks like an issue with Shadow, assuming you aren't selfhosting the link something got messed up with the wisp server it's trying to use.
> 
> i dont even think I have a wisp server, I don't really know how to set up a wisp server and I thought it would set one up automatically
> 
> I am using a VPS run on Ubuntu 22.04, so it's not really self-hosting I think I dunno
> 
> > try going into settings > developer and switch the drop-down on the left to custom. Enter wss://shadow.letsendobesity.org/wisp/ (or put the dropdown back to default) and try again (may need to clear your cache)
> 
> I was thinking of setting up my own wisp server, as that wisp server seems to be getting blocked, I tried a different shadow hosting unblocked on my computer and it worked.
> 
> I also tried it at my home and used it but it still didnt work even with the custom wisp server :C

If you are running it on your own server I highly recommend you run either epoxy-server or the python implementation of wisp, they have much better performance

**Comment by proudparrot2** [comment]
I agree - I'm going to run two load-balanced epoxy-servers on each node in production once we switch over to Wisp
